### PR TITLE
CP-36509 update db for xapi_ssl.pem on startup if changed

### PR DESCRIPTION
**Pull request by lindig**
The host certificate /etc/xensource/xapi-ssl.pem has a corresponding
entry in the xapi database. At startup we check for changes and update
the database entry if the certificate has changed.

The update of the database entry is not yet implemented because the
corresponding code does not yet exist.

Signed-off-by: Christian Lindig <christian.lindig@citrix.com>

### CODE REVIEW

File: ocaml/xapi/certificates_sync.ml [hunk 1]
```diff
@@ -0,0 +1,87 @@
+module D = Debug.Make (struct let name = "certificates_sync" end)
```
Comment (lippirk) [inline ocaml/xapi/certificates_sync.ml:1]: Can we have an mli for this file?

**Comment by lippirk** [comment]
We'll want to sync xapi-pool-tls with the database too, is that going to be done in another PR?

**Comment by psafont** [comment]
I think so, it's not trivial because the startup code is part of the pool join mechanism.

**Review by lippirk — APPROVED** [review]

File: ocaml/xapi/certificates_sync.ml [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+(** determine if the database is up to date by comparing the fingerprint
+  of xapi-ssl.pem with the entry in the database *)
+let is_unchanged ~__context cert_ref cert =
+  let ref_hash = Db.Certificate.get_fingerprint ~__context ~self:cert_ref in
+  let cert_hash =
+    X509.Certificate.fingerprint Mirage_crypto.Hash.(`SHA256) cert
+    |> Certificates.pp_hash
+  in
+  Ok (cert_hash = ref_hash)
```
Comment (psafont) [inline ocaml/xapi/certificates_sync.ml:42]: The function can't return an error, it's worth dropping the result here.
```suggestion
  cert_hash = ref_hash
```

**Review by psafont — APPROVED** [review]

File: ocaml/xapi/certificates_sync.ml (hunk 2 above)
Comment (lindig) [reply ocaml/xapi/certificates_sync.ml]: This needs a change at the caller, too. I've added it.

**Pull request merged**